features/285-design-webhooks/events.md

```diff
@@ -0,0 +1,348 @@
+
+### before_identity_delete, after_identity_delete
+
+When an identity is being deleted for a existing user, e.g. removing login ID/
+unlinking SSO accounts.
+
+```json
+{
+    "identity": { /* an Identity object */ }
+}
+```
+
+### before_session_create, after_session_create
+
+When a session is being created for a existing user, e.g. logging in with
+password/SSO.
+
+```json
+{
+    "reason": "signup",
+    "user": { /* a User object */ },
+    "identity": { /* an Identity object */ }
+}
+```
+
+- `reason`: The reason for the creation of session, can be `signup` or `login`
+
+### before_session_delete, after_session_delete
+
+When a session is being deleted for a existing user, e.g. logging out
+
+```json
+{
+    "reason": "logout",
+    "user": { /* a User object */ },
+    "identity": { /* an Identity object */ }
+}
+```
+
+- `reason`: The reason for the deletion of session, can be `logout`
+
+### before_user_update, after_user_update
+
+When user attributes (metadata, disable status, verified status) is being
+updated due to API operations.
+
+```json
+{
+    "reason": "update-metadata",
+    "is_disabled": true,
+    "is_verified": true,
+    "verify_info": { /* ... */ },
+    "metadata": {},
+    "user": { /* a User object */ }
+}
+```
+
+- `reason`: The reason for the update of user, can be `update-metadata`,
+            `update-identity`, `verification`, or `administrative`.
+- `is_disabled`: The new disabled status;
+                 if it is not changed, the field would be absent.
+- `is_verified`: The new verified status;
+                 if it is not changed, the field would be absent.
+- `verify_info`: The new verify info;
+                 if it is not changed, the field would be absent.
+- `metadata`: The new metadata;
+                 if it is not changed, the field would be absent.
+- `user`: a snapshot of the user object before the operation
+
+**NOTE**
+- This event would not be generated for creation of user.
+  To handle initial values of user, use `before/after_user_create` instead.
+
+### before_password_update, after_password_update
+
+When the password of a user is being updated.
+
+```json
+{
+    "reason": "change-password",
+    "user": { /* a User object */ }
+}
+```
+
+- `reason`: The reason for the update of password, can be `change-password`,
+            `reset-password`, or `administrative`
+- `user`: a snapshot of the user object before the operation
+
+**NOTE**
+- This event would not be generated for creation of user.
+  To handle initial values of user, use `before/after_user_create` instead.
+
+### user_sync
+
+When user state (user/identity/session) is potentially being updated.
+
+```json
+{
+    "user": { /* a User object */ }
+}
+```
+
+- `user`: the user object after the operation
+
+**NOTE**
+- This user object may not be updated for consecutive events; even if the
+  operation does not actually modify the user state, this event would be
+  generated.
+
+
+## Sample Web-hook Event
```

Comment:
The example has the `version` field placed in wrong position. Please correct them.